genome_designer/genome_finish/assembly.py

```diff
@@ -1,169 +1,379 @@
-        opt_dict['velveth']['hash_length'] = kmer_length
-
-        # Run velvet assembly on SV indicants
-        velvet_dir = os.path.join(data_dir, 'velvet_k' + str(kmer_length))
-        run_velvet(
-                SV_indicants_with_pairs_bam,
-                velvet_dir,
-                opt_dict)
-
-        # Collect resulting contigs fasta
-        contigs_fasta = os.path.join(velvet_dir, 'contigs.fa')
-        contig_files.append(contigs_fasta)
-
-        for seq_record in SeqIO.parse(contigs_fasta, 'fasta'):
-
-            contig_label = '_'.join(
-                    [contig_label_base, seq_record.description])
-
-            # Create an insertion model for the contig
-            contig = Contig.objects.create(
-                    label=contig_label,
-                    parent_reference_genome=reference_genome,
-                    experiment_sample_to_alignment=(
-                            experiment_sample_to_alignment))
-
-            contig.metadata['coverage'] = float(
-                    seq_record.description.rsplit('_', 1)[1])
-            contig.metadata['timestamp'] = timestamp
-
-            contig.ensure_model_data_dir_exists()
-
-            dataset_path = os.path.join(contig.get_model_data_dir(),
-                    'fasta.fa')
-            assert not os.path.exists(dataset_path)
-
-            with open(dataset_path, 'w') as fh:
-                SeqIO.write([seq_record], fh, 'fasta')
-
-            add_dataset_to_entity(
-                    contig,
-                    'contig_fasta',
-                    Dataset.TYPE.REFERENCE_GENOME_FASTA,
-                    filesystem_location=dataset_path)
+        pickle.dump(assembly_metadata, fh)
+
+    cmd = ' '.join([
+            VELVETH_BINARY,
+            data_dir,
+            str(velvet_opts['velveth']['hash_length'])] +
+            ['-' + key + ' ' + str(velvet_opts['velveth'][key])
+             for key in velvet_opts['velveth']
+             if key not in ['hash_length']] +
+            ['-bam', '-shortPaired', sv_indicants_bam])
+    print 'velveth cmd:', cmd
+    subprocess.check_call(cmd, shell=True, executable=settings.BASH_PATH)
+
+    ins_length = velvet_opts['velvetg']['ins_length']
+    # exp_cov = velvet_opts['velvetg']['exp_cov']
+    ins_length_sd = velvet_opts['velvetg']['ins_length_sd']
+    cov_cutoff = velvet_opts['velvetg']['cov_cutoff']
+    min_contig_lgth = velvet_opts['velvetg']['min_contig_lgth']
+
+    arg_list = [
+            VELVETG_BINARY,
+            data_dir,
+            '-ins_length', str(ins_length),
+            # '-exp_cov', str(exp_cov),
+            '-ins_length_sd', str(ins_length_sd),
+            '-cov_cutoff', str(cov_cutoff),
+            # '-max_coverage', str(3 * exp_cov),
+            '-min_contig_lgth', str(min_contig_lgth),
+            '-read_trkg', 'yes']
+
+    cmd = ' '.join(arg_list)
+    print 'velvetg cmd:', cmd
+    subprocess.check_call(cmd, shell=True, executable=settings.BASH_PATH)
```

Comment:
maybe add an stderr here to write to a log file in the velvet dir. See for example https://github.com/churchlab/millstone/blob/3180ab2f00f5521ce04b4ae0159f55af94953aaf/genome_designer/pipeline/variant_calling/lumpy.py#L68.